text/0002-numeric-values.md

```diff
@@ -0,0 +1,485 @@
+
+- `Int.min`: The minimum value of this integer type. ex: `i32.min == -2147483648`
+- `Int.max`: The maximum value of this integer type. ex: `u8.max == 255`
+
+## Built-in Properties on Integer Values
+
+All integer values can use property access notation to access built-in properties and methods. The variable name `val` will serve as a placeholder for the integer value used.
+
+- `val.to(T)`: Converts the integer's value to a the given type. T can be
+  - `f64`, which it will convert the value to a [floating point number](#floats)
+  - Another Integer type like `u8`, which it will convert the value to that type. It is undefined behavior if this integer is out of range of `T.min`..`T.max`.
+- `val.pow(y)`: Raise this integer to the power of `y`.
+
+If the integer is a signed integer, the following methods are available:
+
+- `val.abs()`: Get the absolute value of this integer. Returns an unsigned integer type.
+
+## Integer Arithmetic
+
+The following binary operators seen in most other languages are available:
+
+- `+`: Addition
+- `-`: Subtraction
+- `*`: Multiplication
+- `/`: Division
+- `%`: Remainder
+- `<<`: Bitwise left shift
+- `>>`: Bitwise right shift
+- `&`: Bitwise AND
+- `|`: Bitwise OR
+- `^`: Bitwise XOR
+
+The following unary operators are available:
+
+- `-`: Negation
+- `~`: Bitwise NOT
+
+The above operators share the [same precedence as JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence), to reduce confusion.
+
+These all function as JavaScript does, except bitwise operators do not [convert the value to a 32-bit integer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Left_shift#description). Instead, it will keep the number at whatever it's bit width is.
+
+```js
+0 ^ 4294967296;
+// JavaScript -> 0
+// Blight     -> 4294967296
+```
+
+Since fixed-width integer types have a specific minimum and maximum, it is undefined behavior to error to overflow or underflow when doing arithmatic. In a safe build, this will safely crash with a stack trace and information on what values were provided, but at runtime this will cause undefined behavior.
+
+The following operations will cause undefined behavior:
+
+- `+`: If the result of the addition is above `Int.max`
+- `-`: If the result of the subtraction is above `Int.min`.
+- `*`: If the result of the multiplication is above `Int.max` or below `Int.min`.
+- `/`: If there is a remainder when dividing.
+- `/`: If the divisor is `0`.
+- `%`: If the divisor is `0`.
+- Unary `-` can underflow if when negating `Int.max`.
+
+## Floats
+
+There is one floating point type: `f64`, which is a 64-bit floating point that follows the [IEEE 754 standard](https://en.wikipedia.org/wiki/IEEE_754).
+
+## Built-in Properties on the Float Type
+
+`f64` has built in properties. These are accessed by using property access notation.
+
+- `f64.min`: Constant `1.7976931348623157e+308`
+- `f64.max`: Constant `5e-324`
```

Comment:
what...

Reply:
these constants are the largest and smallest values allowed for floating point, as per ieee specification

Reply:
Are they reversed or am I reading this wrong?

Reply:
they are reversed. my mistake. will update the document.